presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java

```diff
@@ -51,4 +45,22 @@ public UserGroupInformation getUserGroupInformation()
         Subject subject = kerberosAuthentication.getSubject();
         return createUserGroupInformationForSubject(subject);
     }
+
+    private Configuration validateHDFSConfiguration(HdfsConfiguration hdfsConfiguration)
+    {
+        Configuration config = hdfsConfiguration.getConfiguration(null, null);
+        String loadedHadoopAuthentication = config.get("hadoop.security.authentication");
+
+        if (!"kerberos".equals(loadedHadoopAuthentication)) {
+            log.warn("Contradicting security authentication type in hive.properties and hive config resources");
+            log.warn("Setting security authentication type to kerberos according to hive.properties");
+            config.set("hadoop.security.authentication", "kerberos");
```

Comment:
why not just `verify("kerberos".equals(config.get("hadoop.security.authentication"))`? 
on the other hand, if this is normal that this is not set, why a warning here?

Reply:
@findepi If the execution arrives at this point in the code, it means that the `HiveAuthenticationModule`  already bound the `kerberosHdfsAuthenticationModule`, after looking at `HdfsAuthenticationType.KERBEROS` in the `HiveClientConfig`. But now, we are using this `HdfsConfiguration` object, which has loaded configs from a different source, mentioned in `hive.config.resources`.  

This check mainly serves as a warning in the case when a user has specified `hive.hdfs.authentication.type=KERBEROS` in etc/catalog/hive.properties, but one of the resources in hive.config.resources carry a contradicting value for `hadoop.security.authentication`, say `SIMPLE`.

I am unsure about how critical the check is. One thing we can do is just to look for what we need. We can get the value for `hadoop.security.auth_to_local`, and not worry about `hadoop.security.authentication` field at all. Should I go ahead and do that? That would remove the need for verify*** method altogether.

Reply:
Hm, yes they can indeed be contradicting. I am leaning towards not bothering about it. 

> That would remove the need for verify*** method altogether.

+1